cvmfs/server/cvmfs_server_util.sh

```diff
@@ -472,21 +472,39 @@ check_upstream_validity() {
 }
 
 
-# ensure that the installed overlayfs is viable for CernVM-FS. Namely, it must
-# be part of the upstream kernel (since 3.18) and recent enough (kernel 4.2)
+# Ensure that the installed overlayfs is viable for CernVM-FS.
 # Note: More details are in CVM-835.
 # @return  0 if overlayfs is installed and viable
+#          1 if it is not viable, and stdout contains a reason
+# This should probably now be called check_overlayfs_viability except
+#   that for backward compatiblity we need to keep the variable that
+#   overrides it, CVMFS_DONT_CHECK_OVERLAYFS_VERSION, and changing the
+#   function name would make the variable name not make sense.
 check_overlayfs_version() {
   [ -z "$CVMFS_DONT_CHECK_OVERLAYFS_VERSION" ] || return 0
-  local scratch_fstype=$(df -T /var/spool/cvmfs | tail -1 | awk {'print $2'})
+  if ! check_overlayfs; then
```

Comment:
This check should be performed before looking into `$CVMFS_DONT_CHECK_OVERLAYFS_VERSION`